### PR TITLE
CRM-16915 - Filter Profile selector in Configure Event - Registration…

### PR DESCRIPTION
**Pull request by jitendrapurohit**
… tab for non-admin users https://issues.civicrm.org/jira/browse/CRM-16915

### CODE REVIEW

**Comment by davecivicrm** [comment]
Test failures are unrelated.

File: CRM/UF/Page/ProfileEditor.php [hunk 1]
```diff
@@ -20,18 +20,26 @@ public static function registerProfileScripts() {
 
     CRM_Core_Resources::singleton()
       ->addSettingsFactory(function () {
+        $ufGroups = civicrm_api3('UFGroup', 'get', array(
+          'sequential' => 1,
+          'is_active' => 1,
+          'rowCount' => 1000, // FIXME
```
Comment (colemanw) [inline CRM/UF/Page/ProfileEditor.php]: I think this should be `'options' => array('limit' => 0)`

Comment (jitendrapurohit) [inline CRM/UF/Page/ProfileEditor.php]: yes, it works, created a new PR at https://github.com/civicrm/civicrm-core/pull/6336

Thanks for the suggestion.

File: CRM/Event/Form/ManageEvent/Registration.php [hunk 2]
```diff
@@ -374,6 +375,9 @@ public static function getProfileSelectorTypes() {
         'Individual',
       ), CRM_Contact_BAO_ContactType::subTypes('Individual'));
     $configs['allowCoreTypes'][] = 'Participant';
+    if (CRM_Core_Permission::check('manage event profiles') && !CRM_Core_Permission::check('administer CiviCRM')) {
+      $configs['usedFor'][] = 'CiviEvent';
+    }
```
Comment (davecivicrm) [inline CRM/Event/Form/ManageEvent/Registration.php]: Add 'CiviEvent_Additional' to $configs['usedFor'] array.

File: CRM/UF/Page/ProfileEditor.php [hunk 3]
```diff
@@ -20,18 +20,26 @@ public static function registerProfileScripts() {
...
+        ));
+        if (CRM_Core_Permission::check('manage event profiles') && !CRM_Core_Permission::check('administer CiviCRM')) {
+          foreach ($ufGroups['values'] as $key => $value) {
+            $ufJoin = CRM_Core_BAO_UFGroup::getUFJoinRecord($value['id']);
+            if (in_array('CiviEvent', $ufJoin)) {
+              $ufGroups['values'][$key]['module'] = 'CiviEvent';
+            }
+          }
+        }
         return array(
```
Comment (davecivicrm) [inline CRM/UF/Page/ProfileEditor.php]: Handle 'CiviEvent_Additional' here as well ??

Comment (jitendrapurohit) [inline CRM/UF/Page/ProfileEditor.php]: created https://github.com/civicrm/civicrm-core/pull/6355 to handle 'CiviEvent_Additional'.

**Pull request merged**